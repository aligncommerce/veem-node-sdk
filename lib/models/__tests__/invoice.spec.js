import InvoiceModel from 'models/invoice'
import merge from 'lodash/merge'
import INVOICE_MOCK from './__mocks__/invoice'

describe('Invoice Model', () => {
  it('should match the snapshot', () => {
    expect(InvoiceModel.schema).toMatchSnapshot()
  })

  it('should define a valid schema', () => {
    expect(InvoiceModel.schema).to.be.validJsonSchema
  })

  it('should filter out invalid properties', () => {
    const invalidInvoiceRequest = merge({}, INVOICE_MOCK, {
      foo: 'bar',
      amount: {
        foo: 'bar',
      },
      payer: {
        foo: 'bar',
      },
    })

    const invoice = new InvoiceModel(invalidInvoiceRequest)
    const actualInvoiceRequest = invoice.request()

    expect(actualInvoiceRequest).to.deep.equal(INVOICE_MOCK)
  })
})
