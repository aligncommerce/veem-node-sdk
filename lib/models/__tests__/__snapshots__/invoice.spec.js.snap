// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Invoice Model request should match the snapshot 1`] = `
Object {
  "properties": Object {
    "amount": Object {
      "properties": Object {
        "amount": Object {
          "type": "number",
        },
        "currency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "attachments": Object {
      "properties": Object {
        "name": Object {
          "type": "string",
        },
        "referenceId": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "ccEmails": Object {
      "items": Object {
        "type": "string",
      },
      "type": "array",
    },
    "clientId": Object {
      "type": "string",
    },
    "dueDate": Object {
      "type": "string",
    },
    "exchangeRate": Object {
      "properties": Object {
        "fromAmount": Object {
          "type": "string",
        },
        "fromCurrency": Object {
          "type": "string",
        },
        "recipientAccountEmail": Object {
          "type": "string",
        },
        "toAmount": Object {
          "type": "string",
        },
        "toCountry": Object {
          "type": "string",
        },
        "toCurrency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "exchangeRateQuoteId": Object {
      "type": "string",
    },
    "externalInvoiceRefId": Object {
      "type": "string",
    },
    "notes": Object {
      "type": "string",
    },
    "payer": Object {
      "properties": Object {
        "businessName": Object {
          "type": "string",
        },
        "countryCode": Object {
          "type": "string",
        },
        "email": Object {
          "type": "string",
        },
        "firstName": Object {
          "type": "string",
        },
        "lastName": Object {
          "type": "string",
        },
        "phone": Object {
          "type": "string",
        },
        "type": Object {
          "enum": Array [
            "Incomplete",
            "Business",
            "Personal",
          ],
          "type": "string",
        },
      },
      "type": "object",
    },
    "purposeOfPayment": Object {
      "type": "string",
    },
  },
  "title": "InvoiceRequest",
  "type": "object",
}
`;

exports[`Invoice Model response should match the snapshot 1`] = `
Object {
  "properties": Object {
    "amount": Object {
      "properties": Object {
        "amount": Object {
          "type": "number",
        },
        "currency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "attachments": Object {
      "properties": Object {},
      "type": "object",
    },
    "ccEmails": Object {
      "items": Object {
        "type": "string",
      },
      "type": "array",
    },
    "claimLink": Object {
      "type": "string",
    },
    "clientId": Object {
      "type": "string",
    },
    "dueDate": Object {
      "type": "string",
    },
    "exchangeRate": Object {
      "properties": Object {
        "expiry": Object {
          "type": "number",
        },
        "fromAmount": Object {
          "type": "number",
        },
        "fromCurrency": Object {
          "type": "string",
        },
        "id": Object {
          "type": "string",
        },
        "rate": Object {
          "type": "number?",
        },
        "timeCreated": Object {
          "type": "string?",
        },
        "toAmount": Object {
          "type": "number",
        },
        "toCurrency": Object {
          "type": "string",
        },
      },
      "type": "object",
    },
    "exchangeRateQuoteId": Object {
      "type": "string",
    },
    "externalInvoiceRefId": Object {
      "type": "string",
    },
    "id": Object {
      "type": "number",
    },
    "notes": Object {
      "type": "string",
    },
    "payer": Object {
      "properties": Object {
        "businessName": Object {
          "type": "string",
        },
        "countryCode": Object {
          "type": "string",
        },
        "email": Object {
          "type": "string",
        },
        "firstName": Object {
          "type": "string",
        },
        "lastName": Object {
          "type": "string",
        },
        "phone": Object {
          "type": "string",
        },
        "type": Object {
          "enum": Array [
            "Incomplete",
            "Business",
            "Personal",
          ],
          "type": "string",
        },
      },
      "type": "object",
    },
    "purposeOfPayment": Object {
      "type": "string",
    },
    "status": Object {
      "type": "string",
    },
    "timeCreated": Object {
      "type": "string",
    },
  },
  "title": "InvoiceResponse",
  "type": "object",
}
`;
